Assignment/Assignment.Tests/PingProcessTests.cs

```diff
@@ -58,65 +59,75 @@ public void Run_CaptureStdOutput_Success()
     public void RunTaskAsync_Success()
     {
         // Do NOT use async/await in this test.
+        PingResult result = Sut.RunTaskAsync("localhost").Result;
         // Test Sut.RunTaskAsync("localhost");
+        AssertValidPingOutput(result);
     }
 
     [TestMethod]
     public void RunAsync_UsingTaskReturn_Success()
     {
         // Do NOT use async/await in this test.
-        PingResult result = default;
+        PingResult result = Sut.RunAsync("localhost").Result;
         // Test Sut.RunAsync("localhost");
         AssertValidPingOutput(result);
     }
 
     [TestMethod]
-#pragma warning disable CS1998 // Remove this
     async public Task RunAsync_UsingTpl_Success()
     {
         // DO use async/await in this test.
-        PingResult result = default;
-
+        PingResult result = await Sut.RunAsync("localhost");
         // Test Sut.RunAsync("localhost");
         AssertValidPingOutput(result);
     }
-#pragma warning restore CS1998 // Remove this
 
 
     [TestMethod]
     [ExpectedException(typeof(AggregateException))]
     public void RunAsync_UsingTplWithCancellation_CatchAggregateExceptionWrapping()
     {
-        
+        CancellationTokenSource source = new();
+        source.Cancel();
+        Task<PingResult> result = Sut.RunAsync("localhost", source.Token);
+        _ = result.Result;
     }
 
     [TestMethod]
     [ExpectedException(typeof(TaskCanceledException))]
     public void RunAsync_UsingTplWithCancellation_CatchAggregateExceptionWrappingTaskCanceledException()
     {
-        // Use exception.Flatten()
+        CancellationTokenSource source = new();
+        source.Cancel();
+        Task<PingResult> result = Sut.RunAsync("localhost", source.Token);
+        try
+        {
+            _ = result.Result;
+        }
+        catch (AggregateException ex)
+        {
+            throw ex.Flatten().InnerException!;

```

Comment:
This works, but it also drops the 
```suggestion
            exception = exception.Flatten();
            ExceptionDispatchInfo.Capture(exception.InnerException ?? exception).Throw();
```
Doing it this way doesn't drop the exception info.